crates/bevy_pbr/src/prepass/prepass.wgsl

```diff
@@ -107,6 +107,12 @@ fn vertex(vertex_no_morph: Vertex) -> VertexOutput {
     // See https://github.com/gfx-rs/naga/issues/2416
     out.instance_index = get_instance_index(vertex_no_morph.instance_index);
 #endif
+#ifdef BASE_INSTANCE_WORKAROUND
+    // Hack: this ensures the push constant is always used, which works around this issue:
+    // https://github.com/bevyengine/bevy/issues/10509
+    // This can be removed when wgpu 0.19 is released
+    out.position.x += f32(get_instance_index(0u)) * 0.00001;
```

Comment:
I see a couple of problems with the approach of modifying the position in this specific way.

Adding an absolute amount could mean that the difference is significant for something at that scale. Perhaps something like *= 1.00001 as a relative change would not be visually noticeable.

Also, reuse of the depth buffer in the main pass with depth comparison function set to equal may fail which could cause artifacts where some fragments are either not or are incorrectly shaded. This exact workaround would need to be applied in exactly the same way in the main pass vertex stage so that the produced clip position and depth are exactly the same.

Reply:
I've resolved this concern by using `min(f32(get_instance_index(0u), 0.0)`. This will work at any scale, which is preferable to both addition and multiplication.

Reply:
Also added the same line to mesh.wgsl to resolve the other part of this comment.